### PR TITLE
Rework lumbermill

### PR DESCRIPTION
**Pull request by Uwuewsky**
#### Summary
Features "Rework lumbermill trading"

#### Purpose of change
https://github.com/CleverRaven/Cataclysm-DDA/blob/99bd819886b07a7197af9ef166d20d3afd904cad/data/json/npcs/lumbermill_employees/TALK_lumbermill_fabricate.json#L4
With the current system the player has to run to the manager every 2-3 hours to buy a new batch. The prices were a little off (no one is going to buy a stick for a dollar).

#### Describe the solution
Buying without waiting has been moved to a trade dialog. A lumbermill manager now also works as a regular merchant and can sell a small amount of leftover lumber. A manager will not buy wood from a player.
lumb1

If a player needs more resources than are offered through trade, they can order a batch through dialog. The waiting time for an order depends on the price, and can be from 1 day to a month, but in a more realistic scenario around 3 days (there are about 15 NPC workers at the lumbermill).
lumb2

The prices are slightly adjusted based on the documentation, but still much higher than what the player can sell it at, since the lumbermill produces these resources itself.

Unnecessary effects have been moved to obsolete.

#### Testing
lumb3
lumb4
lumb5
lumb6
lumb7

#### Additional context
Is there any way to use `u_buy_item` without actually buying the item? Right now the player is given one file as a copy of the contract. Giving zero items doesn't work because `true_eocs` is not executed.

Any suggestions on prices/dialogs are welcome.

### CODE REVIEW

File: data/json/npcs/lumbermill_employees/TALK_lumbermill_fabricate.json [hunk 1]
```diff
@@ -1,208 +1,255 @@
...
-    "global": false,
-    "effect": [
-      { "npc_add_effect": "lumbermill_waiting_for_wood_beam", "duration": "4 h" },
-      { "npc_add_effect": "currently_busy", "duration": "4 h" }
+    "type": "talk_topic",
+    "id": "TALK_LUMBERMILL_FABRICATE_FINISHED",
+    "dynamic_line": {
+      "math": [ "time_since(n_timer_fabricate_waiting, 'unit':'days')", ">=", "n_number_fabricate_wait" ],
+      "yes": "Let me check…  <u_name>…  Yes, all your lumber are ready, you can pick it up.",
+      "no": "Your order?  No, sorry, we still working on it."
```
Comment (Night-Pryanik) [inline data/json/npcs/lumbermill_employees/TALK_lumbermill_fabricate.json:213]:
```suggestion
      "no": "Your order?  No, sorry, we're still working on it."
```

**Comment by GuardianDll** [comment]
\> Is there any way to use u_buy_item without actually buying the item?

u_spawn_item?

**Comment by Maleclypse** [comment]
Are you planning on following Guardian's suggestion or are you ready for this to merge?

**Comment by Uwuewsky** [comment]
@Maleclypse, ready to merge.

**Pull request merged**